packages/grafana-ui/src/components/Table/TableNG/styles.ts

```diff
@@ -0,0 +1,140 @@
+  }),
+  gridNested: css({
+    height: '100%',
+    width: `calc(100% - ${COLUMN.EXPANDER_WIDTH - TABLE.CELL_PADDING * 2 - 1}px)`,
+    overflow: 'visible',
+    marginLeft: COLUMN.EXPANDER_WIDTH - TABLE.CELL_PADDING - 1,
+    marginBlock: TABLE.CELL_PADDING,
+  }),
+  cellNested: css({ '&[aria-selected=true]': { outline: 'none' } }),
+  noDataNested: css({
+    height: TABLE.NESTED_NO_DATA_HEIGHT,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    color: theme.colors.text.secondary,
+    fontSize: theme.typography.h4.fontSize,
+  }),
+  cellActions: css({
+    display: 'none',
+    position: 'absolute',
+    top: 0,
+    margin: 'auto',
+    height: '100%',
+    color: theme.colors.text.primary,
+    background: theme.isDark ? 'rgba(0, 0, 0, 0.7)' : 'rgba(255, 255, 255, 0.7)',
+    padding: theme.spacing.x0_5,
+    paddingInlineStart: theme.spacing.x1,
+  }),
+  cellActionsEnd: css({ left: 0 }),
+  cellActionsStart: css({ right: 0 }),
+  headerRow: css({
+    paddingBlockStart: 0,
+    fontWeight: 'normal',
+    '& .rdg-cell': { height: '100%', alignItems: 'flex-end' },
+  }),
+  displayNone: css({ display: 'none' }),
+  paginationContainer: css({
+    alignItems: 'center',
+    display: 'flex',
+    justifyContent: 'center',
+    marginTop: '8px',
+    width: '100%',
+  }),
+  paginationSummary: css({
+    color: theme.colors.text.secondary,
+    fontSize: theme.typography.bodySmall.fontSize,
+    display: 'flex',
+    justifyContent: 'flex-end',
+    padding: theme.spacing(0, 1, 0, 2),
+  }),
+  menuItem: css({ maxWidth: '200px' }),
+});
+
+export const getFooterStyles = (justifyContent: Property.JustifyContent) => ({
+  footerCellCountRows: css({ display: 'flex', justifyContent: 'space-between' }),
+  footerCell: css({ display: 'flex', justifyContent: justifyContent || 'space-between' }),
+});
+
+export const getHeaderCellStyles = (theme: GrafanaTheme2, justifyContent: Property.JustifyContent) =>
+  css({
+    display: 'flex',
+    gap: theme.spacing(0.5),
+    zIndex: theme.zIndex.tooltip - 1,
+    paddingInline: TABLE.CELL_PADDING,
+    paddingBlockEnd: TABLE.CELL_PADDING,
+    justifyContent,
+    '&:last-child': { borderInlineEnd: 'none' },
+  });
+
+export const getDefaultCellStyles: TableCellStyles = (theme, { textAlign, textWrap, shouldOverflow }) =>
+  css({
+    display: 'flex',
+    alignItems: 'center',
+    textAlign,
+    justifyContent: getJustifyContent(textAlign),
+    minHeight: '100%',
+    ...(textWrap && { whiteSpace: 'pre-line' }),
+    '&:hover, &[aria-selected=true]': {
+      '.table-cell-actions': { display: 'flex' },
+      ...(shouldOverflow && {
+        zIndex: theme.zIndex.tooltip - 2,
+        whiteSpace: 'pre-line',
+        height: 'fit-content',
+        minWidth: 'fit-content',
+      }),
+    },
+    a: {
+      cursor: 'pointer',
+      color: theme.colors.text.link,
+      textDecoration: 'none',
+      '&:hover': { textDecoration: 'underline' },
+    },
+  });
```

Comment:
cuts like 80 lines and 4 arguments from the method - much less scary.

Reply:
One thing that this does change: I do apply this to every cell type now, given that its not prohibitively expensive to do so.